src-java/kilda-persistence-tinkerpop/src/test/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepositoryTest.java

```diff
@@ -91,8 +94,49 @@ public void setUp() {
 
         switchA = createTestSwitch(TEST_SWITCH_A_ID.getId());
         switchB = createTestSwitch(TEST_SWITCH_B_ID.getId());
+        switchC = createTestSwitch(TEST_SWITCH_C_ID.getId());
 
-        assertEquals(2, switchRepository.findAll().size());
+        assertEquals(3, switchRepository.findAll().size());
+    }
+
+    @Test
+    public void findPortToFlowsMap() {
+        createTestFlow(TEST_FLOW_ID, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_2, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_3, switchA, switchB);
+        createTestFlow(TEST_FLOW_ID_4, switchC, switchB);
+
+        Map<Integer, List<Flow>> result = flowRepository.findSwitchFlowsByPort(switchA.getSwitchId(), null);
+
+        assertNotNull(result);
+        assertTrue("The map must contain a key for the port", result.containsKey(PORT_1));
+        assertNotNull("The map must contain a non-null value for the test port", result.get(PORT_1));
+        assertEquals("The map must contain exactly two keys", 2, result.size());
+        assertEquals("The map must contain exactly 3 flows", 3, result.get(PORT_1).size());
+        assertTrue("The map must contain exactly 3 flows",
```

Comment:
Same here, but I think the line 115 is redundant

Reply:
they together checks that there are exactly these 3 flows. If I remove the 115, there might 15 flows, which are not supposed to be there. It is there to make sure the query doesn't grab more than needed